dnf/yum/rpmtrans.py

```diff
@@ -391,9 +391,21 @@ def _scriptError(self, amount, total, key):
         scriptlet_name = rpm.tagnames.get(amount, "<unknown>")
 
         transaction_list = self._extract_cbkey(key)
-        name = transaction_list[0].pkg.name
+        pkg_currenly_being_handled_name = transaction_list[0].pkg.name
 
-        msg = ("Error in %s scriptlet in rpm package %s" % (scriptlet_name, name))
+        if hasattr(key, "name"):
+            pkg_which_owns_this_scriptlet_name = key.name
+        else:
+            pkg_which_owns_this_scriptlet_name = key
+
+        # In case rpm trigger runs scriptlet of some other package
+        if pkg_which_owns_this_scriptlet_name != pkg_currenly_being_handled_name:
+            msg = ("Error in %s scriptlet in rpm package %s triggered by rpm package %s"
+                   % (scriptlet_name, pkg_which_owns_this_scriptlet_name,
+                      pkg_currenly_being_handled_name))
+        else:
+            msg = ("Error in %s scriptlet in rpm package %s"
+                   % (scriptlet_name, pkg_currenly_being_handled_name))
```

Comment:
I do not understand why we do not provide full NEVRA when it is valid and available. Think that you install two kernels in transaction. And one fails but which one?

Reply:
I updated it to print the full NEVRA for the current package which has the information always available.